Enabled CV ops with plain layout

crazydemo opened the pull request:
Enabled OP List:

- Convolution 1D-3D
- Deconvolution 2D-3D
- Pooling 2D-3D (avg / max)
- Elementwise OPs (abs, clip, exp, log, sqrt, round, logsumexp, relu, leaky_relu, tanh, sigmoid, softmax)

Commit:
enable convolution(1d-3d), deconvolution(2d-3d), pooling(2d-3d), some CV activations.

ciyongch commented on src/relay/backend/contrib/dnnl/codegen.cc:
```diff
@@ -473,9 +493,87 @@ class DNNLJSONSerializer : public backend::contrib::JSONSerializer {
+        ICHECK(call->op.as<OpNode>()) << "Not op node";
+      } else if (name == "dnnl.dense_sigmoid") {
+        call = GetRootCall(fn->body.as<CallNode>(), 1, {"nn.dense", "sigmoid"});
+        ICHECK(call->op.as<OpNode>()) << "Not op node";
```
We might split these `if-then-else` clauses into several section for different ops with post ops like `dense/conv`, etc, with different post -op in their own section to improve the readability of the codes?

crazydemo replied:
fixed. split the clauses.

ciyongch commented on src/runtime/contrib/dnnl/dnnl_json_runtime.cc:
```diff
@@ -90,44 +91,95 @@ class DNNLJSONRuntime : public JSONRuntimeBase {
+      {"log", dnnl::algorithm::eltwise_log},
+      {"sqrt", dnnl::algorithm::eltwise_sqrt},
+      {"round", dnnl::algorithm::eltwise_round},
+      {"logsumexp", dnnl::algorithm::eltwise_logsigmoid},
```
Is this mapping exactly the same?

crazydemo replied:
Not exactly. `logsumexp` is supposed to be mapped to `logsigmoid`.

ciyongch commented on src/runtime/contrib/dnnl/dnnl_json_runtime.cc:
```diff
@@ -166,100 +218,88 @@ class DNNLJSONRuntime : public JSONRuntimeBase {
+    dnnl::memory::dims weights_dims = weight_shape;  // {OC, IC, KD, KH, KW}
     if (groups > 1) {
-      weights_dims = {groups, 1, IC / groups, KH, KW};
+      weights_dims = {groups, 1, input_shape[1] / groups};
```
What's the value for `oc` dimension?

crazydemo replied:
replace `1` with `channels / groups`, which equals to `OC`.

ciyongch commented on src/runtime/contrib/dnnl/dnnl_json_runtime.cc:
```diff
@@ -269,24 +309,126 @@ class DNNLJSONRuntime : public JSONRuntimeBase {
+      auto bias_entry = node.GetInputs()[2];
+      BindDNNLMemory(bias_entry, deconv2d_bias_memory);
+    } else {
+      float bias[channels] = {0};
```
no need to create and push this arg when `has_bias=false`?

crazydemo replied:
no need to create and push. fixed.

driazati committed:
Fix JUnit failure reporting (#10121)

* Fix spacing

* Add try..finally everywhere

* trigger ci

* Fix pytest invocations

* Remove junit collection where no files exist

Co-authored-by: driazati <driazati@users.noreply.github.com>

electriclilies committed:
[BUGFIX] Define kTargetPoolReadWriteAccess globally (#10262)

* Fix bug

* Fix whitespace

* lint

* Move the other consts out of PoolInfo

Commit:
[Hexagon] Don't use cmake glob for auto-generated source files (#10259)

* [Hexagon] Don't use cmake glob for auto-generated source files

Glob treats inputs as patterns: if the file with a given name
does not exist (is to be generated later), it won't be added to
the output.

* Restart CI

billishyahao committed:
Add one extra space to improve diagnostic messages (#10268)

Commit:
[RPC] Link in whole archive with BUILD_STATIC_RUNTIME (#10260)

* [RPC] Link in whole archive with BUILD_STATIC_RUNTIME

* Restart CI

pfk-beta committed:
Gitignore work items in jvm and android_rpc (#10253)

Co-authored-by: pfk-beta <this_email_isnot_working@gmail.com>

pfk-beta committed:
fix RPC waiting for device (#10255)

Co-authored-by: pfk-beta <this_email_isnot_working@gmail.com>

yzh119 committed:
Overload get() function for `Optional` type. (#9748)

* upd

* simplify

* upd

* fix

* upd

* fix docstring

driazati committed:
Skip flaky tensorflow tests (#10276)

See #10275

cc @masahi

Co-authored-by: driazati <driazati@users.noreply.github.com>

Lunderberg committed:
[Cuda] Updated bfloat16 math defs. (#10258)

Required to pass `test_cuda_bf16_vectorize_add` in `tests/python/unittest/test_target_codegen_cuda.py`.

Commit:
[TVMC] Add an end_to_end benchmarking argument when benchmarking. (#10256)

* Add an end_to_end benchmarking argument to TVMC run.

* Add command line test.

* Fix comment syntax.

* Set device to cpu if end_to_end is on.

* Tickle CI

Lunderberg committed:
[VTA] Search for libvta_fsim.so in $TVM_LIBRARY_PATH (#10278)

This adds `$TVM_LIBRARY_PATH` to the search directory of
`vta.libinfo.find_libvta`, matching the behavior of
`tvm._ffi.libinfo.find_lib_path`.

electriclilies committed:
[BUGFIX] fix text printer when TVM_LOG_DEBUG is on (#10279)

lhutton1 committed:
[microNPU] Add support for LeakyReLU (#10127)

* [microNPU] Add support for LeakyReLU

Adds support for offloading an int8 Leaky ReLU activation function
to the NPU by legalizing to a LUT.

Change-Id: I63dd5b16a1a2a747b11f15a5b8124810e2ebf491

* refactor LeakyReLUParams to inherit from LutActivationParams

Change-Id: I35b59200b16a7eff1915f771ab6b5d9181d4f3ab

Commit:
update dnnl version from v1.5 to v2.2 (#10266)

yelite committed:
Use `/usr/bin/env bash` in shebang for all scripts under tests (#10277)

* Use /usr/bin/env bash in shebang

This makes scripts executable on system without /bin/bash (NixOS)

* Use `set -e` in script instead of `bash -e` in shebang

Icemist committed:
Add a conversion of individual operations in FQ2I pass. (#10239)

* Add a conversion of individual operations in FQ2I pass.

* apply review comments

* apply review comments 2

prateek9623 committed:
[CMake] add support for find_package (#10097)

* removed include header path, which are invalid

* 1. added target tvm in a cmake export group
2. added cmake package config file

* added Threads as public dependency

* changed temp config file name for better understanding

mehrdadh committed:
[Docker][Hexagon] Add docker file and scripts (#10263)

* Hexagon docker files added

* trigger

driazati committed:
[ci] Mark `test_autotune_conv2d` flaky (#10298)

See #10297

cc @mehrdadh

Co-authored-by: driazati <driazati@users.noreply.github.com>

vinx13 committed:
[TIR] Add software pipelining (#10066)

* [TIR] Add software pipelining

Co-authored-by: Junru Shao <junrushao1994@gmail.com>
Co-authored-by: Xiyou Zhou <xiyou@octoml.ai>
Co-authored-by: Bohan Hou <32121147+spectrometerHBH@users.noreply.github.com>
Co-authored-by: Siyuan Feng <Hzfengsy@sjtu.edu.cn>
Co-authored-by: Hongyi Jin <3231950289@qq.com>
Co-authored-by: Ruihang Lai <lairuihangdongdong@qq.com>

* fix

* fix

* lint

* fix

* format

* doc

* remove print

* lint

* lint

* doc

* Apply suggestions from code review

Co-authored-by: Junru Shao <junrushao1994@gmail.com>

* address comments

* address comments

* refactor FragmentInfo::GetSize

* remove unused

* refactor

* address comments

Co-authored-by: Junru Shao <junrushao1994@gmail.com>
Co-authored-by: Xiyou Zhou <xiyou@octoml.ai>
Co-authored-by: Bohan Hou <32121147+spectrometerHBH@users.noreply.github.com>
Co-authored-by: Siyuan Feng <Hzfengsy@sjtu.edu.cn>
Co-authored-by: Hongyi Jin <3231950289@qq.com>
Co-authored-by: Ruihang Lai <lairuihangdongdong@qq.com>

AndrewZhaoLuo committed:
[FQ2I] Add support for some unary operators (#10273)

* initial commit

* lint

areusch committed:
Add -i option to fix ASF headers to lint scripts. (#10284)

* Add -i option to fix ASF headers to lint scripts.

* address driazati comments

Commit:
[Hexagon] Pass kDLHexagon device when allocating workspace pool on Hexagon (#10289)

Commit:
[Hexagon] Remember to add common sources when building TVMRT for Hexagon (#10290)

areusch committed:
Allow Ctrl+C during docker/lint.sh. (#10291)

driazati committed:
[ci] Mark some ehtosu tests as flaky (#10301)

See #10300

Co-authored-by: driazati <driazati@users.noreply.github.com>

driazati committed:
[microTVM] Add timeouts for CI tests (#10295)

These shouldn't take longer than 5 minutes but since they have to poll they can end up running for a long while (e.g. this failure: https://ci.tlcpack.ai/blue/organizations/jenkins/tvm/detail/main/2534/pipeline).

cc @mehrdadh

Co-authored-by: driazati <driazati@users.noreply.github.com>

driazati committed:
Add flaky test issue template (#10299)

This adds a template so we can report (and label) flaky test issues separately from CI infra problems. This also helps others report flaky tests by pointing them to the relevant documentation.

cc @areusch @denise-k @hpanda-naut @masahi

Co-authored-by: driazati <driazati@users.noreply.github.com>

Lunderberg committed:
[UnitTest] Disable ptx mma tests on unsupported nvcc versions. (#10229)

* [UnitTest] Disable ptx mma tests on unsupported nvcc versions.

- Modified `tvm.contrib.nvcc.get_cuda_version` to return a
  `(major,minor,release)` tuple rather than a float.

- Implemented `tvm.testing.requries_nvcc_version` decorator to specify
  the minimum `(major,minor,release)` version needed to run a unit
  test.

- Applied decorated to unit tests in `test_tir_ptx_mma.py` that fail
  on earlier nvcc versions.

* Fix lint errors.

* Updated a few of the cuda version checks.

* More lint fixes.

* Only compare major/minor in find_libdevice, not release version.

Commit:
[TIR][Schedule] simpilfy compute_at static bound (#10307)

masahi committed:
[CI] Update CPU image to v0.81 (#10305)

Commit:
[Hexagon] Fix getting/setting DMA state (#10288)

* [Hexagon] Fix getting/setting DMA state

The bits [3:0] of the first word of the descriptor (both 16- and 32-byte)
is the DMA state. It must be set to 0 before starting a DMA transaction.

* Restart CI

Raphael-Hao committed:
update (#10306)

Commit:
[TOPI] Add support for groupped conv3d (#9873)

* [TOPI] Add support for groupped conv3d

Change conv3d to use generic conv implementation which supports groupped
convolutions. Also, remove support for non-float16 tensorcore operations
as they cause large degradation in accuracy. Generic conv now supports
autoscheduler.

* correct none check

* add tests for floordiv simplification

* fixed incorrect test for autoscheduler

* formatting

* add groups to winograd

* fix tensorcore

* manually simplify index instead of relying on simplifier

* formatting

* add groups argument to conv3d_ncdhw_winograd_without_weight_transform

* formatting

areusch committed:
Parameterize test_link_params. (#9276)

electriclilies committed:
[Relay] [Virtual Device] Store function result virtual device in virtual_device_ field (#9848)

* VStore function result virtual devices in virtual_device_ field

* Address Mark's 'mega nit'

* Promote function result virtual device to first class

* Add kVirtualDevice

* move kVirtualDevice

* Fix annotation test

* Progress on parsing & printing

* Fix printing of virtual device attribute

* flake

driazati committed:
[ci] Disable flaky microTVM tests (#10313)

See #10312

cc @masahi

Co-authored-by: driazati <driazati@users.noreply.github.com>

driazati committed:
[ci] Disable flaky cmsisnn tests (#10315)

See #10314

cc @masahi

Co-authored-by: driazati <driazati@users.noreply.github.com>

leeexyz committed:
[TE][Fix] Comparison of the output tensor (#9829)

* [TE][Fix] Comparison of the output tensor

* fix hybrid op issue

* fix tensor replacement in schedule ops

* fix compute inline
